backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -0,0 +1,166 @@
+          LOGGER.warn("CollaboratorResourceMigration Exception: {}", ex.getMessage());
+        } else {
+          LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+
+    LOGGER.info("Migration End");
+  }
+
+  private void migrateProjects() {
+    LOGGER.debug("Projects migration started");
+    Long count = getEntityCount(ProjectEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Projects {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ProjectEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ProjectEntity.class);
+        Root<ProjectEntity> root = criteriaQuery.from(ProjectEntity.class);
+
+        CriteriaQuery<ProjectEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ProjectEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ProjectEntity> projectEntities = typedQuery.getResultList();
+
+        if (projectEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (ProjectEntity projectEntity : projectEntities) {
+            userIds.add(projectEntity.getOwner());
+          }
+          LOGGER.debug("Project userId list : " + userIds);
+
+          // Fetch the experiment owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
+          for (ProjectEntity project : projectEntities) {
+            try {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(
+                      userInfoMap.get(project.getOwner()),
+                      project.getWorkspace(),
+                      project.getWorkspace_type());
+              // if projectVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
+              // ignore the CollaboratorType
+              roleService.createWorkspacePermissions(
+                  workspaceDTO.getWorkspaceName(),
+                  Optional.ofNullable(
+                      WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
+                  project.getId(),
+                  Optional.of(Long.parseLong(project.getOwner())),
+                  ModelDBServiceResourceTypes.PROJECT,
+                  CollaboratorTypeEnum.CollaboratorType.READ_WRITE,
+                  project.getProjectVisibility());
```

Comment:
I think here old visibility ORG_SCOPED_PUBLIC should become new ORG_DEFAULT. cc: @conradoverta to check if I'm wrong here

Reply:
And CollaboratorType and canDeploy parameters should have default value for setResource in case of ORG_DEFAULT

Reply:
I was thinking of something like https://github.com/VertaAI/registry/pull/102/files#diff-00d851bf9cc1b4dfa8203d45a8ac9fed69662de44f46e9872a412e8f270e2844R125

Reply:
Changed